src/components/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import styled from '@emotion/styled';
+import * as radixTooltip from '@radix-ui/react-tooltip';
+import { PropsWithChildren } from 'react';
+import { isDefined } from '~common/helpers/types';
+
+export enum TooltipAlignment {
+  Start = 'start',
+  Center = 'center',
+  End = 'end',
+}
+
+export enum TooltipSide {
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+  Left = 'left',
+}
+
+interface Props {
+  align?: TooltipAlignment;
+  content?: React.ReactNode;
+  isOpen?: boolean;
+  side?: TooltipSide;
+}
+
+/* This is the distance between the point of the arrow and the trigger */
+const TOOLTIP_OFFSET = 4;
+
+/* This is the padding between the edge of the tooltip and the arrow. */
+const ARROW_PADDING = 12;
+
+export function Tooltip(props: PropsWithChildren<Props>) {
+  const { align, children, content, isOpen, side } = props;
+
+  /*
+   * Undefined content or empty strings should not trigger a tooltip
+   */
+  if (!isDefined(content) || (typeof content === 'string' && content.length === 0)) {
+    return <>{children}</>;
+  }
+
+  return (
+    <radixTooltip.Root open={isOpen}>
+      <radixTooltip.Trigger asChild>{children}</radixTooltip.Trigger>
+      <radixTooltip.Portal>
+        <TooltipContent
+          align={align}
+          arrowPadding={ARROW_PADDING}
+          side={side}
+          sideOffset={TOOLTIP_OFFSET}>
+          {content}
+          <TooltipArrow />
+        </TooltipContent>
+      </radixTooltip.Portal>
+    </radixTooltip.Root>
+  );
+}
+
+Tooltip.displayName = 'Tooltip';
+
+const TooltipContent = styled(radixTooltip.Content)`
+  border-radius: var(--echoes-border-radius-200);
+  padding: var(--echoes-dimension-space-50) var(--echoes-dimension-space-150);
+  color: white;
+  background-color: var(--echoes-color-background-inverse);
```

Comment:
We also need to set the default typography and to use a design-token the font-color to make it work correctly with dark theme

Reply:
How did I miss that :facepalm: